### PR TITLE
Refactor : Marketing Button

### PR DESCRIPTION
**Pull request by supriyajjadhav**
### Description
Trello task - https://trello.com/c/mSbMJ7X5/144-uag-block-wise-cleanup

### Types of changes
Breaking change

### Checklist:
- [x] My code is tested
- [x] I've included any necessary tests 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/marketing-button/edit.js [hunk 1]
```diff
@@ -2,786 +2,62 @@
...
 
-	let deviceType = __experimentalGetPreviewDeviceType ? __experimentalGetPreviewDeviceType() : null;
+		if ( null !== element && undefined !== element ) {
+			element.innerHTML = styling( props );
+		}
+	}, [ props ] );
+
+	return (
+		<>
+			{ MarketingButtonSettings( props ) }
```
Comment (rohitp-bsf) [inline src/blocks/marketing-button/edit.js:44]: Use JSX way to call the function like `<Settings />`

Use React.memo to reduce extra calls.

Comment (rohitp-bsf) [reply src/blocks/marketing-button/edit.js:44]: Use common naming, `Settings` and `Render`.

File: src/blocks/marketing-button/styling.js [hunk 2]
```diff
@@ -70,144 +69,203 @@ function styling( props ) {
...
 		prefixLineHeightTablet,
 		prefixLineHeightMobile,
-	} = props.attributes
+	} = props.attributes;
 
-	var selectors = {}
-	var tablet_selectors = {}
-	var mobile_selectors = {}
+	let selectors = {};
+	let tablet_selectors = {};
```
Comment (rohitp-bsf) [inline src/blocks/marketing-button/styling.js:75]: Use CamelCasing everywhere for consistency.

**Review by rohitp-bsf — CHANGES_REQUESTED** [review]

File: src/blocks/marketing-button/styling.js (hunk 2 above)
Comment (supriyajjadhav) [reply src/blocks/marketing-button/styling.js]: Updated all the variables

File: src/blocks/marketing-button/edit.js (hunk 1 above)
Comment (supriyajjadhav) [reply src/blocks/marketing-button/edit.js]: Updated

File: src/blocks/marketing-button/styling.js [hunk 3]
```diff
@@ -70,144 +69,203 @@ function styling( props ) {
...
-	styling_css += generateCSS( mobile_selectors, `${id}.uagb-editor-preview-mode-mobile`, true, "mobile" )
+	styling_css += generateCSS(
+		mobileSelectors,
+		`${ id }.uagb-editor-preview-mode-mobile`,
+		true,
+		'mobile'
+	);
 
-	return styling_css
+	return styling_css;
```
Comment (rohitp-bsf) [inline src/blocks/marketing-button/styling.js:268]: Camel case please!

**Review by rohitp-bsf — CHANGES_REQUESTED** [review]

File: src/blocks/marketing-button/styling.js (hunk 3 above)
Comment (supriyajjadhav) [reply src/blocks/marketing-button/styling.js]: Updated

**Review by rohitp-bsf — APPROVED** [review]

File: src/blocks/marketing-button/settings.js [hunk 4]
```diff
@@ -0,0 +1,1296 @@
...
+		/* webpackChunkName: "chunks/marketing-button/typography-control" */ '@Components/typography'
+	)
+);
+const WebfontLoader = lazy( () =>
+	import(
+		/* webpackChunkName: "chunks/marketing-button/web-font-loader-control" */ '@Components/typography/fontloader'
+	)
+);
+
+const svg_icons = Object.keys( UAGBIcon );
```
Comment (RajkiranBagal) [inline src/blocks/marketing-button/settings.js:25]: camelCase variables.

File: src/blocks/marketing-button/settings.js [hunk 5]
```diff
@@ -0,0 +1,1296 @@
...
+							allowReset
+						/>
+					</>
+				) }
+			</PanelBody>
+		);
+	};
+
+	return (
+		<>
```
Comment (RajkiranBagal) [inline src/blocks/marketing-button/settings.js:1279]: Add just one Suspense for whole return.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/marketing-button/settings.js (hunk 5 above)
Comment (supriyajjadhav) [reply src/blocks/marketing-button/settings.js]: Updated return

File: src/blocks/marketing-button/edit.js [hunk 6]
```diff
@@ -2,786 +2,71 @@
...
+		);
 
-	let deviceType = __experimentalGetPreviewDeviceType ? __experimentalGetPreviewDeviceType() : null;
+		if ( null !== element && undefined !== element ) {
+			element.innerHTML = styling( props );
+		}
+	}, [ props ] );
+
+	return (
+		<>
```
Comment (RajkiranBagal) [inline src/blocks/marketing-button/edit.js:50]: Remove unwanted <>

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Review by sandeshjangam — APPROVED** [review]

**Pull request merged**